sources/packages/web/src/services/AppConfigService.ts

```diff
@@ -75,10 +82,75 @@ export class AppConfigService {
   }
 
   async initAuthService(type: ClientIdType) {
+    if (this.authService) {
+      return;
+    }
     if (this._config) {
+      this._authClientType = type;
       this.authService = await AuthService(this._config, type);
     } else {
       throw new Error("Unable to load application: server is not responding");
     }
   }
+
+  roles(): string[] {
+    if (this.authService && this.authService.tokenParsed) {
+      const resourceAccess = this.authService.tokenParsed.resource_access || {};
+      let roles: string[] = [];
+      for (const resourceKey of Object.keys(resourceAccess)) {
+        const resource = resourceAccess[resourceKey];
+        if (resource.roles) {
+          roles = [...roles, ...resource.roles];
+        }
+      }
+      return roles;
+    }
+    return [];
+  }
+
+  authStatus(options: { type: ClientIdType; path: string }): AuthStatus {
+    if (options.type === this._authClientType) {
+      const auth = this.authService?.authenticated || false;
+      if (auth) {
+        // TODO
+        // 1. Role based solution is not working
+        // 2. For Logged in user KeyCloak clients are acting same
+        // 3. Add some more info on token to identify user type or userName for BCeID users
+        // 4. Now Just implementing one temporary work around
+        let validUser = false;
+        if (this.authService?.tokenParsed) {
+          const token = this.authService?.tokenParsed as any;
+          switch (options.type) {
+            case ClientIdType.INSTITUTION: {
+              if (token["bceid_user_name"]) {
```

Comment:
@abschwenker created the below property on the token that could be used instead "bceid_user_name".

![image](https://user-images.githubusercontent.com/61259237/110991825-f6a77700-8329-11eb-87cd-016b69191491.png)

Reply:
DONE